In [1]:
import pandas as pd

df = pd.read_json("../data/raw_products.json")
df.head()


,product_name,price,unit,supplier,location
0,"MF CL LED 207 Indoor Lights, White","₹ 1,600",Piece,Mayfair Lighting LLP,"Noida, Dist. Gautam Buddha Nagar"
1,"Abaj Aluminium Alloy 6063 LED Logo Light, Ligh...","₹ 21,000",Piece,Abaj Interiors Private Limited,"bapunagar, Ahmedabad"
2,"EnDi Deep Junction Led Light, Lighting Color: ...",₹ 50,Piece,N D Electronics,"Odhav, Ahmedabad"
3,"Pure White & Warm White Led Indoor Light, 5 W",₹ 500,Piece,Led Lighting World,Ahmedabad
4,"Led T Bulb - 10 w (Tufaan), 6500K, Cool Daylight",₹ 138,Piece,M R Electronics,"Surajkund Faridabad, Faridabad"


## 📌 Dataset Preview

The dataset was loaded successfully from a JSON file containing scraped product listings
from the IndiaMART B2B marketplace under the LED lighting category.

Each record includes the following attributes:
- **Product Name**
- **Price**
- **Unit**
- **Supplier Name**
- **Supplier Location**

The preview confirms that the data has been ingested correctly and is ready for
cleaning and exploratory analysis.


In [2]:
df.drop_duplicates(inplace=True)
df.dropna(subset=["product_name", "supplier"], inplace=True)

df.info()


<class 'pandas.DataFrame'>
Index: 29 entries, 0 to 55
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   product_name  29 non-null     str  
 1   price         29 non-null     str  
 2   unit          29 non-null     str  
 3   supplier      29 non-null     str  
 4   location      29 non-null     str  
dtypes: str(5)
memory usage: 1.4 KB


## 🧹 Data Cleaning Summary

Duplicate records were removed to ensure that each product listing is unique.
Rows missing critical information such as **product name** or **supplier name**
were also filtered out.

After cleaning:
- The dataset contains **29 valid records**
- All key fields (**p**


In [3]:
df.to_csv("../data/cleaned_products.csv", index=False)


## 💾 Cleaned Data Export

The cleaned dataset was exported to a CSV file for downstream analysis and reuse.

- **File name:** `cleaned_products.csv`
- **Format:** CSV
- **Index:** Excluded

This structured output can be easily consumed by analytics tools, shared with
stakeholders, or used in further data processing and modeling tasks.


In [4]:
df["location"].value_counts().head(10)


location
New Delhi                           7
Noida, Dist. Gautam Buddha Nagar    3
Ahmedabad                           2
Jaipur                              2
bapunagar, Ahmedabad                1
Odhav, Ahmedabad                    1
Surajkund Faridabad, Faridabad      1
Jodhpur Char Rasta, Ahmedabad       1
Lucknow                             1
Pune                                1
Name: count, dtype: int64

## 📍 Supplier Location Distribution

The majority of suppliers are concentrated in a few key regions.

- **New Delhi** has the highest number of suppliers
- **Delhi NCR region** (New Delhi, Noida, Faridabad) contributes a significant share
- Other notable supplier locations include Ahmedabad, Jaipur, Pune, and Lucknow

**Insight:**  
The high concentration of suppliers in Delhi NCR suggests strong manufacturing,
distribution, and demand-side advantages in this region for LED lighting products.


In [5]:
df["product_name"].value_counts().head(10)


product_name
MF CL LED 207 Indoor Lights, White                                 1
Abaj Aluminium Alloy 6063 LED Logo Light, Lighting Color: White    1
EnDi Deep Junction Led Light, Lighting Color: Cool White           1
Pure White & Warm White Led Indoor Light, 5 W                      1
Led T Bulb - 10 w (Tufaan), 6500K, Cool Daylight                   1
HEXAGONE LIGHT FOR COMMERCIAL SPACE LIGHT                          1
Black Led Lights For Gym, Lighting Color: Warm White               1
LED Magnetic Light                                                 1
EL EC500CW LED Light                                               1
Deep Junction Light, 7W                                            1
Name: count, dtype: int64

## 💡 Product Variety Analysis

Each product name appears only once in the dataset, indicating a high level of
product variety and no dominant repeated listings.

**Insight:**  
The absence of repeated product names suggests that suppliers on the platform
offer highly differentiated LED lighting products, likely catering to specific
use cases and customized requirements rather than standardized mass listings.


In [6]:
import re

df["price_numeric"] = df["price"].apply(
    lambda x: float(re.sub(r"[₹,]", "", x)) if pd.notna(x) else None
)

df["price_numeric"].describe()


count       29.000000
mean      1935.344828
std       4152.316404
min         41.000000
25%        138.000000
50%        540.000000
75%       1600.000000
max      21000.000000
Name: price_numeric, dtype: float64

## 💰 Price Distribution Analysis

Product prices were cleaned and converted from string format to numeric values
to enable quantitative analysis.

**Summary statistics:**
- Minimum price: **₹41**
- Maximum price: **₹21,000**
- Median price: **₹540**
- 75% of products are priced below **₹1,600**

**Insight:**  
Most LED lighting products fall within the budget to mid-range price segment.
A small number of high-priced items significantly increase the average price,
indi


## 🧠 Final Summary

This analysis demonstrates an end-to-end data workflow including data collection,
cleaning, exploratory analysis, and insight generation for LED lighting products
on a B2B marketplace.

The results highlight regional supplier concentration, pricing variability, and
high product diversity, providing a strong foundation for further market and
competitive analysis.
